forward/forward.go

```diff
@@ -169,13 +172,6 @@ func (f *Forwarder) cloneRequest(req *http.Request, u *url.URL) (*http.Request,
 	outReq.URL.Scheme = "http"
 	// We need to make sure the host is defined in the URL (not the actual URI)
 	outReq.URL.Host = req.Host
-	// Make sure we define an opaque URL, so the URI is just the path
-	//
-	// Note that EscapedPath() always return the URL-encoded path, so we have
-	// no way to know if the raw URL in req is encoded or not.  Go HTTP client
-	// always encode URL, so do major browsers. That means it's totally safe for
-	// Lantern usage and possibly others.
-	outReq.URL.Opaque = req.URL.EscapedPath()
```

Comment:
Ok I see the original code is a copy of https://github.com/vulcand/oxy/blob/master/forward/fwd.go#L195, with the purpose of dealing with escaped slashes, as what Go's doc says:

> Note that the Path field is stored in decoded form: /%47%6f%2f becomes /Go/. A consequence is that it is impossible to tell which slashes in the Path were slashes in the raw URL and which were %2f. This distinction is rarely important, but when it is, code must not use Path directly.

With the introduce of `RawPath` in Go 1.5, it should no longer be required. Maybe add a test to make sure? Similar to [the test in `oxy`](https://github.com/vulcand/oxy/blob/master/forward/fwd_test.go#L209)